examples/yaml_test.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"os"
+	"os/exec"
+)
+
+const (
+	timeoutSeconds = 600
+	sleepBetween   = 10
+)
+
+// cmd will run the command c with args and if input is provided, that will be piped
+// into the process as input
+func cmd(t *testing.T, c string, args []string, input string) (string, error) {
+	t.Helper()
+	binary, err := exec.LookPath(c)
+	if err != nil {
+		t.Fatalf("couldn't find %s, plz install", c)
+	}
+
+	t.Logf("Executing %s %v", binary, args)
+
+	cmd := exec.Command(binary, args...)
+	cmd.Env = os.Environ()
+
+	if input != "" {
+		cmd.Stdin = strings.NewReader(input)
+	}
+
+	var stderr, stdout bytes.Buffer
+	cmd.Stderr = &stderr
+	cmd.Stdout = &stdout
+	if err := cmd.Run(); err != nil {
+		t.Fatalf("couldn't run command %s %v: %v, %s", c, args, err, stderr.String())
+		return "", err
+	}
+
+	return stdout.String(), nil
+}
+
+// resetState will erase all types that the yaml examples could have concievably created.
+func resetState(t *testing.T) {
+	t.Helper()
+
+	crdTypes := []string{
+		"conditions.tekton.dev",
+		"pipelineresources.tekton.dev",
+		"tasks.tekton.dev",
+		"clustertasks.tekton.dev",
+		"pipelines.tekton.dev",
+		"taskruns.tekton.dev",
+		"pipelineruns.tekton.dev",
+		"services",
+		"pods",
+		"configmaps",
+		"secrets",
+		"serviceaccounts",
+		"persistentvolumeclaims",
+	}
+	for _, c := range crdTypes {
```

Comment:
Eventually we should be able to parse the YAML file and create the resource using a client, instead of involving kubectl.

For this specifically we can just use the CRD client to delete the types.

It'd be fine to have a TODO for that for now but that's a better final state I think.

Reply:
Yes and no. We will need to also test some `kubectl create` commands as this is something the user will do.

Reply:
I'm not 100% sure which is better - if we want to use the clients, we need to load the yamls + find find the right client for the thing being created. i think im like 60% convinced this is better in the long run, but there's also something nice about just blasting it all out and watching it. i do think the part where i look for the word "run" in the ko output is pretty hacky, ill put a comment in about that for sure